app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt

```diff
@@ -89,12 +82,14 @@ class DefaultBrowserToolbarController(
     internal var ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO)
 
     override fun handleToolbarPaste(text: String) {
-        adjustBackgroundAndNavigate.invoke(
-            BrowserFragmentDirections.actionBrowserFragmentToSearchFragment(
+        captureEngineViewAndDrawStatically.invoke {
+            val directions = BrowserFragmentDirections.actionBrowserFragmentToSearchFragment(
```

Comment:
This is conceptually fine, but I find it a little difficult to read as is.  It would be nice if this could look more like

`captureEngineViewAndDrawStatically.andThen { nav }`, or `captureEngineViewAndDrawStatically.doOnEnd { nav }`